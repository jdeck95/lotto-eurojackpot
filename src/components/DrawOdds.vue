<script setup lang="ts">
import type { Odd } from '@/types/eurojackpot';
import { ref } from 'vue';

interface Props {
  odds: Odd[];
}

const props = defineProps<Props>();

const columns = [
  { name: 'winningClass', required: true, label: 'Gewinnklasse', field: 'winningClass' },
  { name: 'numberOfWinners', required: true, label: 'Anzahl Gewinner', field: 'numberOfWinners' },
  { name: 'oddsInCent', required: true, label: 'Gewinnsumme', field: 'odd' },
];

const pagination = ref({
  rowsPerPage: 0,
});
</script>

<template>
  <q-table
    :rows="props.odds"
    :columns="columns"
    v-model:pagination="pagination"
    row-key="name"
    flat
    hide-pagination
  />
</template>
