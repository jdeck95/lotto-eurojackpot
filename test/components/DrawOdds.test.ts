import { mount, type VueWrapper } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import { setActivePinia } from 'pinia';
import { Quasar } from 'quasar';
import { beforeEach, describe, expect, it } from 'vitest';
import DrawOdds from '../../src/components/DrawOdds.vue';

describe('DrawOdds', () => {
  let wrapper: VueWrapper<any>;

  const pinia = createTestingPinia();
  setActivePinia(pinia);

  beforeEach(() => {
    wrapper = mount(DrawOdds, {
      global: {
        plugins: [Quasar, pinia],
      },
      props: {
        odds: [
          {
            numberOfWinners: '12',
            winningClass: 1,
            odd: '123456',
          },
        ],
      },
    });
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
