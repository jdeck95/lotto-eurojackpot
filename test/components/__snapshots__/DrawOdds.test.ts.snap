// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DrawOdds > should render correctly 1`] = `
"<div class="q-table__container q-table--horizontal-separator column no-wrap q-table__card q-table--flat q-table--no-wrap">
  <!---->
  <div class="q-table__middle scroll">
    <table class="q-table">
      <thead>
        <tr>
          <th class="text-right">Gewinnklasse</th>
          <th class="text-right">Anzahl Gewinner</th>
          <th class="text-right">Gewinnsumme</th>
        </tr>
      </thead>
      <tbody>
        <tr class="">
          <td class="text-right">1</td>
          <td class="text-right">12</td>
          <td class="text-right">123456</td>
        </tr>
      </tbody>
    </table>
  </div>
  <!---->
</div>"
`;
